applications/FluidDynamicsApplication/custom_elements/vms_adjoint_element.h

```diff
@@ -366,9 +366,115 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) VMSAdjointElement: public Element
     {
         KRATOS_TRY
 
-        KRATOS_ERROR << "this function is not implemented.";
+        // Check sizes and initialize matrix
+        if (rLeftHandSideMatrix.size1() != TFluidLocalSize ||
+            rLeftHandSideMatrix.size2() != TFluidLocalSize)
+            rLeftHandSideMatrix.resize(TFluidLocalSize, TFluidLocalSize, false);
 
-        KRATOS_CATCH("")
+        noalias(rLeftHandSideMatrix) = ZeroMatrix(TFluidLocalSize, TFluidLocalSize);
+
+        if (rRightHandSideVector.size() != TFluidLocalSize)
+            rRightHandSideVector.resize(TFluidLocalSize, false);
+
+        noalias(rRightHandSideVector) = ZeroVector(TFluidLocalSize);
+
+        // Calculate RHS
+        // Calculate this element's geometric parameters
+        double gauss_weight;
+        array_1d<double, TNumNodes> N;
+        BoundedMatrix<double, TNumNodes, TDim> DN_DX;
+        GeometryUtils::CalculateGeometryData(this->GetGeometry(), DN_DX, N, gauss_weight);
+
+        // Calculate this element's fluid properties
+        double density;
+        array_1d<double, TDim> body_force;
+        FluidCalculationUtilities::EvaluateInPoint(this->GetGeometry(), N,
+                                                   std::tie(density, DENSITY),
+                                                   std::tie(body_force, BODY_FORCE));
+
+        double coeff = density * gauss_weight;
+
+        // Add the results to the velocity components (Local Dofs are vx, vy, [vz,] p for each node)
+        int local_index = 0;
+
+        for (unsigned int i_node = 0; i_node < TNumNodes; ++i_node) {
+            for (unsigned int d = 0; d < TDim; ++d) {
+                rRightHandSideVector[local_index++] +=
+                    coeff * N[i_node] * body_force[d];
+            }
+            ++local_index; // Skip pressure Dof
+        }
+
+        // For OSS: Add projection of residuals to RHS
+        if (rCurrentProcessInfo[OSS_SWITCH] == 1) {
+            KRATOS_ERROR
+                << "OSS projections are not yet supported with VMS adjoints.\n";
+        }
```

Comment:
I'd check this only once in the  `Check` function.

Reply:
done :)